agent/http.go

```diff
@@ -359,8 +359,9 @@ func parseWait(resp http.ResponseWriter, req *http.Request, b *structs.QueryOpti
 
 // parseConsistency is used to parse the ?stale and ?consistent query params.
 // Returns true on error
-func parseConsistency(resp http.ResponseWriter, req *http.Request, b *structs.QueryOptions) bool {
+func (s *HTTPServer) parseConsistency(resp http.ResponseWriter, req *http.Request, b *structs.QueryOptions) bool {
 	query := req.URL.Query()
+	b.AllowStale = !b.RequireConsistent && s.agent.config.HttpApiConfig.AllowStale
```

Comment:
My original suggestion was to simplify that boolean logic because I found it hard to read, but I didn't see the full diff, so missed the fact that RequireConsistent was not yet set. Now that @wojtkiewicz has rearranged code to set RequireConsistent before, this line `b.AllowStale = !b.RequireConsistent && s.agent.config.HttpApiConfig.AllowStale` , is doing the right thing. The flow is: First initialize allowStale according to whether the httpApiConfig.allowStale has it set (AND if b.RequireConsistent is not set, since you shouldn't have allowStale/allowConsistent both be true). 
Then in the next line, it potentially overrides the value of allowStale if the query param contains "stale" 
+1 to your suggestion of failing with a 400 if both are set . It looks like [line 371](https://github.com/hashicorp/consul/pull/3142/files#diff-03f1adb0cf098d94755af53cf0b484d6R371) is already doing that, and in the right place.